src/control/server/ctl_storage_rpc.go

```diff
@@ -306,3 +306,57 @@ func (c *ControlService) StorageNvmeRebind(ctx context.Context, req *ctlpb.NvmeR
 
 	return resp, nil
 }
+
+// StorageNvmeAddDevice adds a newly added SSD to a DAOS engine's NVMe config to allow it to be used.
+func (c *ControlService) StorageNvmeAddDevice(ctx context.Context, req *ctlpb.NvmeAddDeviceReq) (resp *ctlpb.NvmeAddDeviceResp, err error) {
+	c.log.Debugf("received StorageNvmeAddDevice RPC %v", req)
+
+	if req == nil {
+		return nil, errors.New("nil request")
+	}
+
+	engines := c.harness.Instances()
+	engineIndex := req.GetEngineIndex()
+	if len(engines) <= int(engineIndex) {
+		return nil, errors.Errorf("engine with index %d not found", engineIndex)
+	}
+	defer func() {
+		err = errors.Wrapf(err, "engine %d", engineIndex)
+	}()
+
+	engineStorage := engines[engineIndex].GetStorage()
+
+	bdevCfgs := engineStorage.GetBdevConfigs()
+	if len(bdevCfgs) == 0 {
+		return nil, errors.New("no bdev storage tiers in config")
+	}
+
+	tierIndex := req.GetBdevTierIndex()
+	if len(bdevCfgs) <= int(tierIndex) {
+		return nil, errors.Errorf("bdev tier with index %d not found", tierIndex)
+	}
+	tierCfg := bdevCfgs[tierIndex]
```

Comment:
The thinking is that the user would specify which Bdev tier to insert the device into, there isn't an obvious way to detect which tier to automatically insert into AFAIK. This basically says use the new plugged SSD into the Nth bdev tier.

Reply:
BdevTierIndex isn't currently set in the command parameters and so it will default to the first (0 index) Bdev tier. It can be enabled as a user specified parameter later and I can change it to not be bdev specific then.

Reply:
fixed